File: frontend/src/types/aiInsights.ts
import type { CategoryBreakdownItem } from "./analytics";

export interface CategoryBreakdownInsight {
  status: "ready";
  items: CategoryBreakdownItem[];
}

/** Every section WIREFRAMES.md names that the backend doesn't build ships
 * this shape instead of a fabricated list — `blocked_reason` is always a
 * real, specific explanation (API_SPEC.md §14). There is deliberately no
 * `items` field here, so a component can't mistake "not built" for
 * "no results". */
export interface BlockedInsightSection {
  status: "blocked";
  blocked_reason: string;
}

export interface AIInsightsResponse {
  category_breakdown: CategoryBreakdownInsight;
  trending_issues: BlockedInsightSection;
  repeated_problems: BlockedInsightSection;
  high_risk_alerts: BlockedInsightSection;
  recommendations: BlockedInsightSection;
}

File: backend/app/schemas/ai_insights.py
from typing import Literal

from pydantic import BaseModel

from app.schemas.analytics import CategoryBreakdownItem


class CategoryBreakdownInsight(BaseModel):
    status: Literal["ready"] = "ready"
    items: list[CategoryBreakdownItem]


class BlockedInsightSection(BaseModel):
    """Every AI Insights section WIREFRAMES.md names but this change doesn't
    build renders this shape -- `status` is always "blocked" and there is no
    `items` field, so a client can't accidentally render an empty list as
    "no results" when the truth is "not implemented, see blocked_reason"
    (same discipline as KpiValue in schemas/analytics.py)."""

    status: Literal["blocked"] = "blocked"
    blocked_reason: str


class AIInsightsResponse(BaseModel):
    category_breakdown: CategoryBreakdownInsight
    trending_issues: BlockedInsightSection
    repeated_problems: BlockedInsightSection
    high_risk_alerts: BlockedInsightSection
    recommendations: BlockedInsightSection
